### PR TITLE
[CP Staging] Fix #27673

### LINKED ISSUE #27673: [HOLD for payment 2023-09-29] [HOLD for payment 2023-09-27] Web - Wrong navigation after back from add-bank-account page
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Click on your avatar to open settings
2. Click wallet > Then Add payment method > Then Bank account
3. Now Click on the browser back button

## Expected Result:
User should return to the wallet page

## Actual Result:
The user did not return to the wallet page, but instead went back to the settings page.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.71.4
**Reproducible in staging?:** y
**Reproducible in production?:** n
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

[Screencast from 17-09-23 04_21_16.webm](https://github.com/Expensify/App/assets/93399543/bd1f4291-8409-4d15-a6d6-1aa056c6f3f3)

https://github.com/Expensify/App/assets/93399543/75d385cc-dd74-4a1a-b4bb-1bfbda858f67

**Expensify/Expensify Issue URL:**
**Issue reported by:** @Ahmed-Abdella
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1694914706789169

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01993d4f2d40318905</li>
        <li>Upwork Job ID: 1710721784525074432</li>
        <li>Last Price Increase: 2023-10-07</li>
    </ul>
</details>

### PR DESCRIPTION
**Pull request by ginsuma**
### Details
Reset Plaid steps if the screen is blurred.

### Fixed Issues

$ https://github.com/Expensify/App/issues/23888
$ https://github.com/Expensify/App/issues/27673
PROPOSAL: https://github.com/Expensify/App/issues/23888#issuecomment-1657243651

### Tests
1. Open the app
2. Open settings
3. Open workspaces and open any workspace
4. Open bank account
5. Click on 'Connect online with plaid' and while loading, click on get assistance icon in header.
6. Verify that the app doesn't open the next step of 'Connect online with plaid'.
7. Open settings
8. Open Wallet
9. Add a bank account.
10. Waiting PlaidLink and click to close button.
11. Verify that the app goes back to the wallet page. 
- [x] Verify that no errors appear in the JS console

### Offline tests

N/A
### QA Steps
Same as tests

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/13113013/4337a848-8c5a-4874-bc88-c9507b421288

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/13113013/b2fcf7cc-f918-40e8-a8c2-d86403c22a61

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/13113013/e4315d12-3e26-4152-af61-2e236e68082c

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/13113013/80ec028f-ce4d-4e8b-9f51-1c38166fdcd1

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/13113013/8db48631-8d99-41c5-9978-a3cca634cc63

</details>

<details>

<summary>Android</summary>

https://github.com/Expensify/App/assets/13113013/2b05b010-4c66-46f5-a887-01c19d5c0a76

</details>

### CODE REVIEW

File: src/pages/ReimbursementAccount/BankAccountPlaidStep.js [hunk 1]
```diff
@@ -51,6 +53,14 @@ function BankAccountPlaidStep(props) {
         return errorFields;
     }, []);
 
+    useEffect(() => {
```
Comment (joelbettner) [inline src/pages/ReimbursementAccount/BankAccountPlaidStep.js:56]: I'm not entirely sure how this works.

So, any time the Plaid Setup element is focused, or any time the `plaidData` gets updated we will run the `useEffect` method, correct?

What about when the element is unfocused (like on a blur)?

**Review by joelbettner — APPROVED** [review]

**Comment by joelbettner** [comment]
@abdulrahuman5196 all yours for review and checklist.

**Comment by thienlnam** [comment]
Assigning @aimane-chnaif since they can get to this now and it's a deploy blocker

**Review by aimane-chnaif — APPROVED** [review]
This PR
-fully reverts https://github.com/Expensify/App/pull/26816 to fix https://github.com/Expensify/App/issues/27673 which is deploy blocker
-and then add reset logic on blur of BankAccountPlaidStep to fix https://github.com/Expensify/App/issues/23888 which was original issue

**Review by thienlnam — APPROVED** [review]

**Pull request merged**